### PR TITLE
add performance options and minor fixes for 3.1

### PR DESCRIPTION
**Pull request by alknopfler**
This PR contains the following changes:
- Minor fixes for the versions numbers in 3.1 (needed to create the release 3.1 in the repo) related to the downstream clusters.
- Add some performance scripts after internal work improving the performance for 5g apps. After this PR official docs will be updated, with the improvements
- Modify the sriov script to reduce the capi manifest complexity because it's static file and it could be included in eib custom files.

### CODE REVIEW

**Comment by alknopfler** [comment]
- mgmt-cluster tested in lab properly with 3.1 and all features working fine

File: telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml [hunk 1]
```diff
@@ -114,77 +114,28 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/sriov-network-operator-chart
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:115]:
```suggestion
                    chart: oci://registry.suse.com/edge/3.1/sriov-network-operator-chart
```

File: telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml [hunk 2]
```diff
@@ -100,7 +100,7 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/sriov-crd-chart
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:101]:
```suggestion
                    chart: oci://registry.suse.com/edge/3.1/sriov-crd-chart
```

File: telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml [hunk 3]
```diff
@@ -100,7 +100,7 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/sriov-crd-chart
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:101]:
```suggestion
                    chart: oci://registry.suse.com/edge/3.1/sriov-crd-chart
```

File: telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml [hunk 4]
```diff
@@ -114,77 +114,28 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/sriov-network-operator-chart
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:115]:
```suggestion
                    chart: oci://registry.suse.com/edge/3.1/sriov-network-operator-chart
```

File: telco-examples/edge-clusters/dhcp-less/eib/custom/scripts/04-sriov.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# copy the artifacts
+cp sriov-auto-filler.sh /var/sriov-auto-filler.sh
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/dhcp-less/eib/custom/scripts/04-sriov.sh:4]: We need to `mount /var` first and unmount it after.

File: telco-examples/edge-clusters/dhcp/eib/custom/scripts/04-sriov.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# copy the artifacts
+cp sriov-auto-filler.sh /var/sriov-auto-filler.sh
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/dhcp/eib/custom/scripts/04-sriov.sh:4]: I might have missed this but where is this script used?

[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp/eib/custom/scripts/04-sriov.sh:4]: I was assuming it was done internally in EIB when you're joining all scripts. It was working without mount /var in 3.0...

Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp/eib/custom/scripts/04-sriov.sh:4]: telco-examples/edge-clusters/airgap/eib/custom/files/sriov-auto-filler.sh

File: telco-examples/edge-clusters/dhcp-less/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml [hunk 7]
```diff
@@ -102,7 +102,7 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/metallb-chart
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp-less/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml:103]: changed

File: telco-examples/edge-clusters/dhcp-less/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml [hunk 8]
```diff
@@ -88,7 +88,7 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/endpoint-copier-operator-chart
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp-less/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml:89]: changed

File: telco-examples/edge-clusters/dhcp/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml [hunk 9]
```diff
@@ -88,7 +89,7 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/endpoint-copier-operator-chart
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml:109]: changed

File: telco-examples/edge-clusters/dhcp/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml [hunk 10]
```diff
@@ -102,7 +103,7 @@ spec:
                   spec:
                     chart: oci://registry.suse.com/edge/metallb-chart
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml:123]: changed

File: telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml (hunk 1 above)
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:115]: changed

File: telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml (hunk 2 above)
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp-less/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:101]: changed

File: telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml (hunk 3 above)
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:101]: changed

File: telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml (hunk 4 above)
Comment (alknopfler) [reply telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node-sriov-auto.yaml:115]: changed

File: telco-examples/edge-clusters/airgap/eib/custom/scripts/04-sriov.sh [hunk 11]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# copy the artifacts
+cp sriov-auto-filler.sh /var/sriov-auto-filler.sh
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/eib/custom/scripts/04-sriov.sh]: changed to /opt

File: telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node.yaml [hunk 12]
```diff
@@ -237,35 +236,32 @@ spec:
...
                 Wants=network-online.target
-                After=network.target network-online.target
+                After=network.target network-online.target cpu-partitioning.service
                 [Service]
                 Type=oneshot
                 User=root
-                ExecStart=/bin/sh -c "echo isolated_cores=${ISOLATED_CPU_CORES} > /etc/tuned/cpu-partitioning-variables.conf"
-                ExecStartPost=/bin/sh -c "tuned-adm profile cpu-partitioning"
-                ExecStartPost=/bin/sh -c "systemctl enable tuned.service"
+                ExecStart=/bin/sh -c /opt/performance-settings.sh
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/dhcp/edge-telco-single-node/telco-capi-single-node.yaml:264]:
```suggestion
                ExecStart=/bin/sh -c /opt/performance-settings/performance-settings.sh
```

**Review by atanasdinov — APPROVED** [review]

File: telco-examples/edge-clusters/README.md [hunk 13]
```diff
@@ -138,11 +137,10 @@ The first thing is to modify the `telco-capi-single-node.yaml` file and replace
...
-- `${RKE2_VERSION}` - The RKE2 version to be used for the edge cluster. For example, `v1.28.3+rke2r1` could be used for the edge cluster.
 
 You can also modify any other parameter in the `telco-capi-single-node.yaml` file to match with your requirements e.g. DPDK configuration, number of VFs to generate, number of SRIOV interfaces, etc. This is basically a template to be used for the edge cluster deployment. 
 
-** Note: Remember to locate the `eibimage-slemicro55rt-telco.raw` file generated in [Create the image for the edge cluster](#create-the-image-for-the-edge-cluster) into the management cluster httpd cache folder to be used during the edge cluster provision step.
+** Note: Remember to locate the `eibimage-slmicro60rt-telco.raw` file generated in [Create the image for the edge cluster](#create-the-image-for-the-edge-cluster) into the management cluster httpd cache folder to be used during the edge cluster provision step.
```
Comment (hardys) [inline telco-examples/edge-clusters/README.md:143]: We don't have to change it in this PR, but in future we could rename this e.g just to `eibimage.raw` or similar to reduce the things we need to update in future

**Review by hardys — APPROVED** [review]

File: telco-examples/edge-clusters/README.md (hunk 13 above)
Comment (hardys) [reply telco-examples/edge-clusters/README.md:143]: Although actually if we want to enable multiple versions in the imagecache perhaps we need to keep this convention 🤔

File: telco-examples/edge-clusters/dhcp/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml [hunk 14]
```diff
@@ -76,6 +77,25 @@ spec:
                 WantedBy=multi-user.target
         storage:
           files:
+            # https://docs.rke2.io/networking/multus_sriov#using-multus-with-cilium
```
Comment (hardys) [inline telco-examples/edge-clusters/dhcp/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml:80]: Do we need to add this to the other examples too e.g single-node?

**Comment by alknopfler** [comment]
I missed to add the comment about testing:
- rke2-server configured properly
- cmdline args updated with the performance improvmenets:
```
localhost:/var/lib/rancher/rke2/server/manifests # cat /proc/cmdline
skew_tick=1 BOOT_IMAGE=/boot/vmlinuz-6.4.0-9-rt root=UUID=3770e41b-dabf-483c-bcc3-71fdbdc71f00 rd.timeout=60 rd.retry=45 console=ttyS0,115200 console=tty0 security=selinux selinux=1 quiet net.ifnames=0 ignition.platform.id=openstack net.ifnames=1 intel_iommu=on iommu=pt idle=poll mce=off hugepagesz=1G hugepages=40 hugepagesz=2M hugepages=0 default_hugepagesz=1G irqaffinity=0-3 isolcpus=domain,nohz,managed_irq,4-39 nohz_full=4-39 rcu_nocbs=4-39 rcu_nocb_poll nosoftlockup nowatchdog nohz=on nmi_watchdog=0 skew_tick=1
```
- performance-settings systemd executed properly:
```
localhost:/var/lib/rancher/rke2/server/manifests # journalctl -u performance-settings.service
Oct 18 11:34:11 localhost.localdomain systemd[1]: Starting performance-settings...
Oct 18 11:34:11 localhost.localdomain sh[8252]: Configure: CpuFreq performance
Oct 18 11:34:11 localhost.localdomain sh[8252]: Configure: Disable Timer migration
Oct 18 11:34:11 localhost.localdomain sh[8353]: kernel.timer_migration = 0
Oct 18 11:34:11 localhost.localdomain sh[8252]: Configure: Migrate kthreads to HK
Oct 18 11:34:11 localhost.localdomain sh[8362]: pid 464's current affinity list: 0-39
Oct 18 11:34:11 localhost.localdomain sh[8362]: pid 464's new affinity list: 0-39
Oct 18 11:34:11 localhost.localdomain sh[8368]: pid 449's current affinity list: 0-39
Oct 18 11:34:11 localhost.localdomain sh[8368]: pid 449's new affinity list: 0-39
Oct 18 11:34:11 localhost.localdomain sh[8252]: Configure: IsolCpus latency requirements
Oct 18 11:34:11 localhost.localdomain sh[8252]: WARNING: Using idle=poll as a kernel paramter makes per-cpu pm qos redundant
Oct 18 11:34:11 localhost.localdomain sh[8252]: Configure: Delay vmstat updates
Oct 18 11:34:11 localhost.localdomain sh[8372]: vm.stat_interval = 300
Oct 18 11:34:11 localhost.localdomain systemd[1]: performance-settings.service: Deactivated successfully.
Oct 18 11:34:11 localhost.localdomain systemd[1]: Finished performance-settings.
```
- sriov script automatic vf creation completed successfully:
```
localhost:/var/lib/rancher/rke2/server/manifests # cat configmap-sriov-custom-auto.yaml resource1.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: sriov-custom-auto-config
  namespace: sriov-network-operator
data:
  config.json: |
    [
       {
         "resourceName": "resource1",
         "interface": "ens4f0np0",
         "pfname": "ens4f0np0",
         "driver": "vfio-pci",
         "numVFsToCreate": 2
       }
    ]
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetworkNodePolicy
metadata:
  name: atip-resource1
  namespace: sriov-network-operator
spec:
  nodeSelector:
    feature.node.kubernetes.io/network-sriov.capable: "true"
  resourceName: resource1
  deviceType: vfio-pci
  numVfs: 2
  mtu: 1500
  nicSelector:
    pfNames: ["ens4f0np0"]
    deviceID: "1593"
    vendor: "8086"
    rootDevices:
      - 0000:43:00.0
      
localhost:/etc/systemd/system # dpdk-devbind.py -s

Network devices using DPDK-compatible driver
============================================
0000:43:01.0 'Ethernet Adaptive Virtual Function 1889' drv=vfio-pci unused=iavf
0000:43:01.1 'Ethernet Adaptive Virtual Function 1889' drv=vfio-pci unused=iavf

Network devices using kernel driver
===================================
0000:16:00.0 'Ethernet Controller X550 1563' if=eno1 drv=ixgbe unused=vfio-pci *Active*
0000:16:00.1 'Ethernet Controller X550 1563' if=eno2 drv=ixgbe unused=vfio-pci
0000:43:00.0 'Ethernet Controller E810-C for SFP 1593' if=ens4f0np0 drv=ice unused=vfio-pci
0000:43:00.1 'Ethernet Controller E810-C for SFP 1593' if=ens4f1np1 drv=ice unused=vfio-pci
0000:43:00.2 'Ethernet Controller E810-C for SFP 1593' if=ens4f2np2 drv=ice unused=vfio-pci
0000:43:00.3 'Ethernet Controller E810-C for SFP 1593' if=ens4f3np3 drv=ice unused=vfio-pci
0000:70:00.0 'Ethernet Controller E810-C for SFP 1593' if=ens6f0np0 drv=ice unused=vfio-pci
0000:70:00.1 'Ethernet Controller E810-C for SFP 1593' if=ens6f1np1 drv=ice unused=vfio-pci
0000:70:00.2 'Ethernet Controller E810-C for SFP 1593' if=ens6f2np2 drv=ice unused=vfio-pci
0000:70:00.3 'Ethernet Controller E810-C for SFP 1593' if=ens6f3np3 drv=ice unused=vfio-pci
```

**Pull request merged**